cmd/collector/app/sanitizer_v2/service_name_sanitizer.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package sanitizer_v2
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// Cache interface similar to the one in V1
+type Cache interface {
+	Get(alias string) string
+	IsEmpty() bool
+}
+
+// serviceNameSanitizer sanitizes the service names in span annotations given a source of truth alias to service cache.
+type serviceNameSanitizer struct {
```

Comment:
we don't need this file in v2. There is a chance that Uber is still using it, but they will have to fine a different way of integrating this functionality, e.g. they can implement a custom processor.

Please add a Compatibility section to cmd/jaeger/README.md and mention this there, we should keep track of these small things.